modules/dnn/perf/perf_layer.cpp

```diff
@@ -707,4 +707,65 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_FullyConnected, Combine(
     dnnBackendsAndTargets()
 ));
 
+struct Layer_Softmax : public TestBaseWithParam<tuple<Backend, Target> >
+{
+    void test_layer(const std::vector<int>& shape, int axis)
+    {
+        int backendId = get<0>(GetParam());
+        int targetId = get<1>(GetParam());
+
+        Mat data(shape, CV_32FC1);
+
+        Scalar mean = 0.f;
+        Scalar std = 1.f;
+        randn(data, mean, std);
+
+        Net net;
+        LayerParams lp;
+        lp.type = "Softmax";
+        lp.name = "testLayer";
+        lp.set("axis", axis);
+
+        int id = net.addLayerToPrev(lp.name, lp.type, lp);
+        net.connect(0, 0, id, 0);
+
+        // warmup
+        {
+            std::vector<String> inpNames(1);
+            inpNames[0] = "data";
+            net.setInputsNames(inpNames);
+            net.setInput(data, inpNames[0]);
+
+            net.setPreferableBackend(backendId);
+            net.setPreferableTarget(targetId);
+            Mat out = net.forward();
+        }
+
+        TEST_CYCLE()
+        {
+            Mat res = net.forward();
+        }
+
+        SANITY_CHECK_NOTHING();
+    }
+};
+
+
+PERF_TEST_P_(Layer_Softmax, Softmax_small)
+{
+    test_layer({16, 50, 50}, 2);
+}
+
+PERF_TEST_P_(Layer_Softmax, Softmax_middle)
+{
+    test_layer({16, 197, 197}, 2);
+}
+
+PERF_TEST_P_(Layer_Softmax, Softmax_large)
+{
+    test_layer({16, 1080, 1920, 3}, 2);
+}
```

Comment:
Make sense to add a parameter for different axes to `INSTANTIATE_TEST_CASE_P`

Reply:
Test all axis cases is needed, but the axis may be different for different dimension.

Reply:
Should I just make the perf test to a same dimension?

Reply:
Yep, the same dimensions if fine. I mean varying axis might be also useful:
```
{16, 1080, 1920, 3}, 1
{16, 1080, 1920, 3}, 2
{16, 1080, 1920, 3}, 3
```
Instead of a new tests, you may add a parameter

Reply:
Thanks, I will do it in next commit.